core/sql/arkcmp/CmpContext.cpp

```diff
@@ -1157,4 +1157,16 @@ void CmpContext::resetLogmxEventSqlText()
    delete sqlTextBuf_ ;
    sqlTextBuf_ = NULL ;
 }
+
+void CmpContext::clearAllCaches()
+{
+   qcache_->makeEmpty();
```

Comment:
The constructor for CmpContext seems to initialize histogramCache_ the same as qcache_. Is it possible qcache_ can be NULL at some point? It may be safer to guard it, this is a really minor nit to be handled later if necessary.

There is one more cache, optPCodeCache_ . Now this cache may not cause any priv violations I suppose. So a comment later here may be all we need, that PCode cache is not addressed by this method.